Issue 137

javerom opened the pull request:
Fixes #137 

- Lock button from `Settings`
- On lock, returns to main toolbar and main window; menu is replaced with `HOLD TO UNLOCK`
- Must hold the button for X ms until the screen unlocks
- X is defined in `default_settings`

javerom commented:
Haven't yet implemented the numpad, but shouldn't take long to do so.

carmelopellegrino commented:
Hi Javier! Just checked out the branch and found the following problems:
- the main ui is still fully clickable even if the screen is locked (for example the alarm status of monitors can be toggled off)
- some times (random) the gui crashes with the following exception when the `LOCK Screen` button is pressed:

```
Traceback (most recent call last):
  File "/home/ccc/Work/mvm/gui/gui/mainwindow.py", line 285, in lock_screen
    self.set_screenlock_state(True)
  File "/home/ccc/Work/mvm/gui/gui/mainwindow.py", line 277, in set_screenlock_state
    self.button_unlockscreen.pressed.disconnect()
TypeError: disconnect() failed between 'pressed' and all its connections
Aborted (core dumped)
```

javerom commented:
Fixed crashing errors and implemented numpad lockout after long press of Unlock button. Behavior is now:
- Click LOCK SCREEN -> Screen is disabled and returns to main window with HOLD TO LOCK instead of Menu
- Long press HOLD TO LOCK (configurable ms) -> A number pad shows up in the bottom bar. Back returns to locked screen, inputting the correct code unlocks the screen.
- The code only needs to be inputted correctly once (i.e. only the last N digits input are considered as part of the input code)
